### PR TITLE
Add tracing logs for fee policy creation

### PR DESCRIPTION
**Pull request by sunce86**
# Description
Adding some trace logs for fee policies creation.

There are some orders for which I can't figure out reliably why they are considered in-market without these logs.

### CODE REVIEW

**Review by squadgazzz — APPROVED** [review]

File: crates/autopilot/src/domain/fee/mod.rs [hunk 1]
```diff
@@ -43,9 +43,11 @@ impl ProtocolFee {
                 // if the quote is missing, we can't determine if the order is outside the
                 // market price so we protect the user and not charge a fee
                 let Some(quote) = quote else {
+                    tracing::trace!(?order.metadata.uid, "quote is missing, not charging a fee");
```
Comment (fleupold) [inline crates/autopilot/src/domain/fee/mod.rs:46]: This strikes me as something we should at least debug log (I thought we even had a warning log somewhere when there is a missing quote). I think we actively need to debug those cases and make sure that every order that makes it into the system has a code.

File: crates/autopilot/src/domain/fee/mod.rs [hunk 2]
```diff
@@ -43,9 +43,11 @@ impl ProtocolFee {
...
                     return vec![];
                 };
 
+                tracing::trace!(?order.metadata.uid, ?order.data.sell_amount, ?order.data.buy_amount, ?quote, "checking if order is outside market price");
```
Comment (fleupold) [inline crates/autopilot/src/domain/fee/mod.rs:50]: May as well log the fee, no?

**Review by fleupold — APPROVED** [review]

File: crates/autopilot/src/domain/fee/mod.rs (hunk 1 above)
Comment (sunce86) [reply crates/autopilot/src/domain/fee/mod.rs:46]: We have it here https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/database/quotes.rs#L50-L56
I didn't know should I add this one also, but did it for completness.

Comment (fleupold) [reply crates/autopilot/src/domain/fee/mod.rs:46]: If we already have this information elsewhere, let's not log it twice. While I think we could still do a better job at adding more important debugging info, we should also not overlog the same data in multiple places.

Comment (sunce86) [reply crates/autopilot/src/domain/fee/mod.rs]: removed

**Pull request merged**